books/projects/shnf/top.lisp

```diff
@@ -22,15 +22,28 @@
 
 ;; A polynomial term over a list of variables:
 
+(defn distinct-symbols (vars)
```

Comment:
Move distinct-symbols here to use in guard of `polyp`.
Add `null` test because guard of `member` requires `true-listp`.